src/samplers/multinomial.jl

```diff
@@ -1,6 +1,6 @@
 
-function multinom_rand!(n::Int, p::AbstractVector{Float64},
-                        x::AbstractVector{T}) where T<:Real
+function multinom_rand!(n::Int, p::AbstractVector{PT},
```

Comment:
`PT` is not needed 
```suggestion
function multinom_rand!(n::Int, p::AbstractVector{<:Real},
```
Actually also `T` is not needed, replace the lines below by
```
    if i == km1
        @inbounds x[k] = n
    else  # n must have been zero
        for j = i+1 : k
            @inbounds x[j] = 0
        end
    end
```